includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -776,19 +777,24 @@ protected function getArticlesDetails( $articleIds, $articleKeys = [], $width =
 		$collection = $this->appendMetadata( $collection );
 
 		$thumbnails = null;
-		//if strict return to original ids order
+		$collection = $this->preserveOriginalOrder( $articleIds, $collection );
+		//if strict return to original ids order - as an array
 		if ( $strict ) {
-			foreach( $articleIds as $id ) {
-				if ( !empty( $collection[ $id ] ) ) {
-					$result[] = $collection[ $id ];
-				}
-			}
-			return $result;
+			return array_values( $collection );
 		}
-
 		return $collection;
```

Comment:
This is protected method used in 4 places of this class. Some API-s like (getList, getMostLinked) need ordered array, others - like getDetails - need dictionary. If we want to keep the output of those APIs unchanged - we need to have the possibility of returning results in both formats (dict and array). However we still can sort the dict version to keep the responses consistent.